### ISSUE TITLE
undefined method `authenticate_user!' when want to version my api

### ISSUE THREAD

**Issue by Mirigan** [lynndylanhurley/devise_token_auth#908]
Hello ! 
I have a huge problem, to day I wanted to versioning my api and my auth routes so I did this 
```
namespace :api do
    namespace :v1 do
      mount_devise_token_auth_for 'User', as: 'v1', at: 'auth', controllers: {
        registrations:  'api/v1/user/registrations',
        passwords:      'api/v1/user/passwords',
        sessions:       'api/v1/user/sessions'
      }
    end

    namespace :v2 do
      mount_devise_token_auth_for 'User', as: 'v2', at: 'auth', controllers: {
        registrations:  'api/v2/user/registrations',
        passwords:      'api/v2/user/passwords',
        sessions:       'api/v2/user/sessions'
      }
    end
  end
```

but now I have an error `undefined method 'authenticate_user!'` and don't find solution.
Can someone have any idea ?

**Comment by gundogantekant** [comment]
As a quick work-around I moved my mounting code to top, out of namespaces since both versions will use the same credentials. Then I realized it is possible to have multiple same entry. When you just add
`mount_devise_token_auth_for 'User', as: 'v1', at: 'auth'` 
to top, devise_auth will see which model you are working with.

Also amitfriedman12 [wrote](https://github.com/lynndylanhurley/devise_token_auth/issues/182#issuecomment-171197712) in this issue using devise helper to your both controllers might help. 

`include Devise::Controllers::Helpers`
